### ISSUE TITLE
Error with pre-create check: "Hyper-V PowerShell Module is not available"

### ISSUE THREAD

**Issue by rogermartensson** [docker/machine#4342]
Version:      17.09.

Command:
docker-machine create -d hyperv --hyperv-virtual-switch "myswitch" name1
Running pre-create checks...
Error with pre-create check: "Hyper-V PowerShell Module is not available"

I found problem in: https://github.com/docker/machine/blob/master/drivers/hyperv/powershell.go
The reason for the problem is the Check is looking for Get-VM. If you have VMware PowerCLI installed it can happen that the test will find the VMware version of the Get-VM command. And this isn't correct and will fail.

The solution is to add the modulename to the Get-VM name.

**Comment by Wicaeed** [comment]
Currently affected by this issue. Is there a workaround?

**Comment by cljk** [comment]
Hmmm.. I´m getting the same error... but I have the PowerShell Module installed and have NO Vmware CLI installed.... 

Any ideas!?

**Comment by cljk** [comment]
Okay... I think I´m affected from issue #4342 - differenct cause (docker for windows installed)

**Comment by capnchamomile** [comment]
Same issue here. I can run `Get-VM` and it will list VMs in Hyper-V but when I attempt to run `docker-machine create -d hyperv --hyperv-virtual-switch "myswitch" myvm1` I get the "Hyper-V Powershell Module is not available" error

**Comment by SuperFlea2828** [comment]
Same issue. This started happening after the upgrade from 17.x to 18.03.0-ce-win59 (16762). Get-VM is also working. Running Win 10 Pro version 1803 OS build 17128.1

docker-machine create -d hyperv --hyperv-virtual-switch "myswitch" myvm1
Running pre-create checks...
Error with pre-create check: "Hyper-V PowerShell Module is not available"

Get-VM
Name        State   CPUUsage(%) MemoryAssigned(M) Uptime           Status             Version
----        -----   ----------- ----------------- ------           ------             -------
MobyLinuxVM Running 0           2048              02:10:08.2950000 Operating normally 8.3

**Comment by cljk** [comment]
I now switched back to Virtualbox and removed Hyper-V completely. The whole network management of Hyper-V was a pain independent of the problems with docker-machine.

**Comment by DrDoa** [comment]
I would use a hyper-v linux instead using docker directly

**Comment by sincerekamal** [comment]
I'm using `docker-machine version 0.14.0, build 89b8332` in my macbook, but still I get this error, any update on fixing this issue ?

**Comment by MikyChow** [comment]
i had the same issue now,is any one can help us fix it?
![image](https://user-images.githubusercontent.com/5389478/38679689-f0bc4436-3e96-11e8-8fc5-a3756de336cc.png)

**Comment by thangchung** [comment]
Trying to downgrade `minikube` to `0.25.0`, and it should work. 
You can find the issues at https://github.com/kubernetes/minikube/issues/2634
I used `Windows 10`, then just simply run the command as below
`choco install minikube --version 0.25.0 --allow-downgrade`
Hope this helps while we're waiting for a fixed version.

**Comment by cljk** [comment]
Dont know if its the same with minikube - but you cant successfully install docker with chocolatey. At least some weeks ago there was a bug in choco install routine

**Comment by IamGabrielWu** [comment]
is this problem fixed ? i still got the problem.
```
PS C:\Users\yingp>  docker-machine create --driver hyperv DC2-N1
Running pre-create checks...
Error with pre-create check: "Hyper-V PowerShell Module is not available"
PS C:\Users\yingp>
```

**Comment by flashdoggy** [comment]
@kelvinnguyen2310  Holly F*k S* !!!!!! YOU!!!! SAVED!!! MY!!!! LIFE!!!!
I've been stuck here for a almost whole day!!!!! You!! Just!!! Saved!!! Me!!!! 
even just been downgrade the version of the docker-machine.

**Comment by vparam** [comment]
docker-machine to v0.15.0 has fixed the issue. Thanks folks

**Comment by fadyboy** [comment]
I'm still having this issue with docker-machine v0.15.0

**Comment by webstandardcss** [comment]
# Here is the solution

### ` https://github.com/docker/machine/releases/download/v0.15.0/docker-machine-Windows-x86_64.exe`

### Save the downloaded file to your existing directory containing `docker-machine.exe`.   

For my system this is the location for `docker-machine.exe`
`/c/Program Files/Docker/Docker/Resources/bin/docker-machine.exe`

### Backup the old file and replace it file with the new one. 

`cp docker-machine.exe docker-machine.014.exe`

### Rename the downloaded filename to `docker-machine.exe`  

`mv docker-machine-Windows-x86_64.exe docker-machine.exe`

## Build Instructions

1. Create virtual switch in Hyper-V manager named `myswitch`
2. Request Docker to create a VM named `myvm1`

`docker-machine create -d hyperv --hyperv-virtual-switch "myswitch" myvm1`

## Results
`docker-machine create -d hyperv --hyperv-virtual-switch "myswitch" myvm1`

~~~~log
Running pre-create checks...
(myvm1) Image cache directory does not exist, creating it at C:\Users\Trey Brister\.docker\machine\cache...
(myvm1) No default Boot2Docker ISO found locally, downloading the latest release...
(myvm1) Latest release for github.com/boot2docker/boot2docker is v18.05.0-ce
(myvm1) Downloading C:\Users\Trey Brister\.docker\machine\cache\boot2docker.iso from https://github.com/boot2docker/boot2docker/releases/download/v18.05.0-ce/boot2docker.iso...
(myvm1) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
Creating machine...
(myvm1) Copying C:\Users\Trey Brister\.docker\machine\cache\boot2docker.iso to C:\Users\Trey Brister\.docker\machine\machines\myvm1\boot2docker.iso...
(myvm1) Creating SSH key...
(myvm1) Creating VM...
(myvm1) Using switch "myswitch"
(myvm1) Creating VHD
(myvm1) Starting VM...
(myvm1) Waiting for host to start...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with boot2docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: C:\Program Files\Docker\Docker\Resources\bin\docker-machine.exe env myvm1
~~~~

**Comment by hinews** [comment]
Solved in 0.15 version https://github.com/docker/machine/tree/v0.15.0

**Comment by hinews** [comment]
I got another error with 0.13 and 0.15
Starting VM...
Error creating machine: Error in driver during machine creation: exit status 1
